### PR TITLE
Added XML Comments

### PR DESCRIPTION
**Pull request by nicolastakashi**
Resolve issue #19

### CODE REVIEW

File: Flepper.QueryBuilder/Commands/Extensions/FromCommandExtensions.cs [hunk 1]
```diff
@@ -1,16 +1,43 @@
...
+        /// Add Inner Join to query
+        /// </summary>
+        /// <param name="fromCommand">From command instance</param>
+        /// <param name="table">Table name</param>
+        /// <returns></returns>
         public static IJoin InnerJoin(this IFromCommand fromCommand, string table)
             => fromCommand.To<Join>().InnerJoin(table);
 
+        /// <summary>
+        /// Add Inner Join to query
```
Comment (gustavoferrazfontes) [inline Flepper.QueryBuilder/Commands/Extensions/FromCommandExtensions.cs]: add **left join** to query not inner

File: Flepper.QueryBuilder/Commands/Extensions/SelectCommandExtensions.cs [hunk 2]
```diff
@@ -1,13 +1,36 @@
...
+        /// <param name="table">Table name</param>
+        /// <returns></returns>
+        public static IFromCommand From(this ISelectCommand selectCommand, string table)
             => selectCommand.To((s, p) => new FromCommand(s, p, table));
 
+        /// <summary>
+        /// Add Top Command
+        /// </summary>
+        /// <param name="selectCommand">Select Command instance</param>
+        /// <param name="size">Quantity of columns</param>
```
Comment (gustavoferrazfontes) [inline Flepper.QueryBuilder/Commands/Extensions/SelectCommandExtensions.cs]: size is quantity of record.

File: Flepper.QueryBuilder/Commands/Interfaces/IQueryCommand.cs [hunk 3]
```diff
@@ -4,12 +4,35 @@
...
     public interface IQueryCommand
     {
+        /// <summary>
+        /// Build a query
+        /// </summary>
+        /// <returns></returns>
         string Build();
+
+        /// <summary>
+        /// Build a query
```
Comment (gustavoferrazfontes) [inline Flepper.QueryBuilder/Commands/Interfaces/IQueryCommand.cs]: Build a query using parameters

File: Flepper.QueryBuilder/Commands/Interfaces/IQueryCommand.cs [hunk 4]
```diff
@@ -4,12 +4,35 @@
...
+        /// </summary>
+        /// <returns></returns>
         QueryResult BuildWithParameters();
 
+        /// <summary>
+        /// Map Query to nem Command
```
Comment (gustavoferrazfontes) [inline Flepper.QueryBuilder/Commands/Interfaces/IQueryCommand.cs]: map query to  **a** Command

File: Flepper.QueryBuilder/Commands/Interfaces/IQueryCommand.cs [hunk 5]
```diff
@@ -4,12 +4,35 @@
...
+        /// </summary>
+        /// <typeparam name="TEnd"></typeparam>
+        /// <returns></returns>
         TEnd To<TEnd>() where TEnd : IQueryCommand;
+
+        /// <summary>
+        /// Map Query to nem Command
```
Comment (gustavoferrazfontes) [inline Flepper.QueryBuilder/Commands/Interfaces/IQueryCommand.cs]: map query to a Command

File: Flepper.QueryBuilder/Join/Interfaces/IJoin.cs [hunk 6]
```diff
@@ -1,8 +1,22 @@
 ﻿namespace Flepper.QueryBuilder
 {
+    /// <summary>
+    /// Join Interfaced
```
Comment (gustavoferrazfontes) [inline Flepper.QueryBuilder/Join/Interfaces/IJoin.cs]: Join **Interface**

File: Flepper.QueryBuilder/Operators/Alias/Extensions/AliasOperatorExtensions.cs [hunk 7]
```diff
@@ -1,17 +1,45 @@
...
+        /// <param name="table">Column Name</param>
+        /// <returns></returns>
         public static IJoin InnerJoin(this IAliasOperator aliasOperator, string table)
             => aliasOperator.To<Join>().InnerJoin(table);
 
+        /// <summary>
+        /// Add Left Join Operator to query
+        /// </summary>
+        /// <param name="aliasOperator">Alias Operator instance</param>
+        /// <param name="table">Column Name</param>
```
Comment (gustavoferrazfontes) [inline Flepper.QueryBuilder/Operators/Alias/Extensions/AliasOperatorExtensions.cs]: Table Name

File: Flepper.QueryBuilder/Operators/Comparison/Interfaces/IComparisonOperators.cs [hunk 8]
```diff
@@ -1,12 +1,50 @@
...
 {
+    /// <summary>
+    /// Comparison Operator Interface
+    /// </summary>
     public interface IComparisonOperators : IQueryCommand
     {
+        /// <summary>
+        /// Equal Comparison Operator Contract
+        /// </summary>
+        /// <param name="value">Value</param>
```
Comment (gustavoferrazfontes) [inline Flepper.QueryBuilder/Operators/Comparison/Interfaces/IComparisonOperators.cs]: a value using to compare

File: Flepper.QueryBuilder/Operators/Comparison/Interfaces/IComparisonOperators.cs [hunk 9]
```diff
@@ -1,12 +1,50 @@
...
+        /// <returns></returns>
         IComparisonOperators EqualTo(object value);
+
+        /// <summary>
+        /// Greater Than Comparison Operator Contract
+        /// </summary>
+        /// <param name="value">Value</param>
```
Comment (gustavoferrazfontes) [inline Flepper.QueryBuilder/Operators/Comparison/Interfaces/IComparisonOperators.cs]: a value using to compare

File: Flepper.QueryBuilder/Operators/Comparison/Interfaces/IComparisonOperators.cs [hunk 10]
```diff
@@ -1,12 +1,50 @@
...
+        /// <returns></returns>
         IComparisonOperators GreaterThan(int value);
+
+        /// <summary>
+        /// Less Than Comparison Operator Contract
+        /// </summary>
+        /// <param name="value">Value</param>
```
Comment (gustavoferrazfontes) [inline Flepper.QueryBuilder/Operators/Comparison/Interfaces/IComparisonOperators.cs]: a value using to compare

File: Flepper.QueryBuilder/Operators/Comparison/Interfaces/IComparisonOperators.cs [hunk 11]
```diff
@@ -1,12 +1,50 @@
...
+        /// Less Than Or Equal Comparison Operator Contract
+        /// </summary>
+        /// <param name="value">Value</param>
+        /// <returns></returns>
         IComparisonOperators LessThanOrEqualTo(int value);
+
+        /// <summary>
+        /// Not equal Comparison Operator Contract
+        /// </summary>
+        /// <param name="value">Value</param>
```
Comment (gustavoferrazfontes) [inline Flepper.QueryBuilder/Operators/Comparison/Interfaces/IComparisonOperators.cs]: a value using to compare

File: Flepper.QueryBuilder/Operators/Values/Interfaces/IValuesOperator.cs [hunk 12]
```diff
@@ -1,7 +1,15 @@
...
 {
+    /// <summary>
+    /// Values Operator Interface
+    /// </summary>
     public interface IValuesOperator : IQueryCommand
     {
+        /// <summary>
+        /// Values Operator Contract
+        /// </summary>
+        /// <param name="values">Values</param>
```
Comment (gustavoferrazfontes) [inline Flepper.QueryBuilder/Operators/Values/Interfaces/IValuesOperator.cs]: a array of values

File: Flepper.QueryBuilder/Commands/Extensions/SelectCommandExtensions.cs [hunk 13]
```diff
@@ -1,13 +1,36 @@
...
+        /// <param name="table">Table name</param>
+        /// <returns></returns>
+        public static IFromCommand From(this ISelectCommand selectCommand, string table)
             => selectCommand.To((s, p) => new FromCommand(s, p, table));
 
+        /// <summary>
+        /// Add Top Command
+        /// </summary>
+        /// <param name="selectCommand">Select Command instance</param>
+        /// <param name="size">Size of records</param>
```
Comment (AlbertoMonteiro) [inline Flepper.QueryBuilder/Commands/Extensions/SelectCommandExtensions.cs]: How about **Quantity of records**?

File: Flepper.QueryBuilder/Commands/Extensions/UpdateCommandExtensions.cs [hunk 14]
```diff
@@ -2,14 +2,38 @@
...
+    /// <summary>
+    /// Update Command Extension
+    /// </summary>
     public static class UpdateCommandExtensions
     {
+        /// <summary>
+        /// Add Set to query
+        /// </summary>
+        /// <param name="updateCommand">Update Command instance</param>
+        /// <param name="column">Column name</param>
```
Comment (AlbertoMonteiro) [inline Flepper.QueryBuilder/Commands/Extensions/UpdateCommandExtensions.cs]: How about **Column to update**?

File: Flepper.QueryBuilder/Commands/Extensions/UpdateCommandExtensions.cs [hunk 15]
```diff
@@ -2,14 +2,38 @@
...
+        /// <param name="value">Value to update</param>
```
Comment (AlbertoMonteiro) [inline Flepper.QueryBuilder/Commands/Extensions/UpdateCommandExtensions.cs]: How about **Value to apply**?

File: Flepper.QueryBuilder/Commands/Extensions/UpdateCommandExtensions.cs (hunk 14 above)
Comment (nicolastakashi) [reply Flepper.QueryBuilder/Commands/Extensions/UpdateCommandExtensions.cs]: Yes! For exemple: UPDATE Table SET Column = Value.

We have a Table name a Column name that will be updated and a new value to update.

File: Flepper.QueryBuilder/Commands/Extensions/UpdateCommandExtensions.cs (hunk 15 above)
Comment (nicolastakashi) [reply Flepper.QueryBuilder/Commands/Extensions/UpdateCommandExtensions.cs]: The same of the last comment.

**Pull request merged**